### PR TITLE
docs: create documentation outline

### PR DESCRIPTION
**Pull request by rishkwal**

### CODE REVIEW

**Review by mojoX911 — COMMENTED** [review]
Comments

File: v1/0_introduction.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Introduction to Coinswap
+
+Coinswap is a protocol in Bitcoin that enables trustless, decentralized coin swaps between parties without requiring intermediaries. Originally proposed by Greg Maxwell in 2013 on bitcointalk.org, the protocol has evolved significantly, with Chris Belcher introducing important improvements in 2020 through the bitcoin-dev mailing list.
```
Comment (mojoX911) [inline v1/0_introduction.md:3]: Exapnd on the Chris Belcher part a bit more.

- He gave a complete design covering the edge cases. In the mailing lists. https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2020-October/018221.html
- a more detailed design doc mentioning future improvements. https://gist.github.com/chris-belcher/9144bd57a91c194e332fb5ca371d0964
- Then he built a Proof of Concept in rust. https://github.com/bitcoin-teleport/teleport-transactions
- He couldn't continue further due to health issues. The project got stale on June 2021. https://gist.github.com/chris-belcher/ca5051285c6f8d38693fd127575be44d
- We picked up where he left and carried on the building. This is still visible in our main repo as "forked from" tag. https://github.com/citadel-tech/coinswap 

This project is a continuation of belcher's legacy. We will finish what he started.

File: v1/0_introduction.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+## What is Coinswap?
+
+At its core, Coinswap is an intra-chain atomic swap protocol, meaning it facilitates the exchange of bitcoin between different parties within the same blockchain. Unlike cross-chain atomic swaps that enable exchanges between different cryptocurrencies, Coinswap focuses on improving privacy and fungibility within the Bitcoin network itself. The protocol's key innovation lies in its atomic nature – transactions either complete fully or fail entirely, eliminating the risks of partial trades or asset loss during the exchange process. This atomicity is achieved through sophisticated time-locked contracts and cryptographic techniques.
```
Comment (mojoX911) [inline v1/0_introduction.md:9]: This needs modification. We ARE going for cross-chain atomic swaps. This makes it sounds like we we are something different. Leave this part on me I will open a PR on this soon.

**Review by mojoX911 — APPROVED** [review]
Ack. I am okay with just merging this and doing improvements on followup PRs. Below are listed few. There could be many more. I will work on the introduction page and improve the copy a bit. 

@rishkwal feel free to turn any of these suggests below into issues, and tag relevant people. Or if you feel some small changes can be done before merge can do that too.

File: v1/0_introduction.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (rishkwal) [reply v1/0_introduction.md]: Fixed

File: v1/6_privacy.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# Privacy
+
+Coinswap was proposed as a privacy-enhancing protocol for Bitcoin transactions. This section explores the privacy features of Coinswap, including its impact on traceability, linkability, fungibility, confidentiality, anonymity, and privacy trade-offs. It also discusses privacy considerations, best practices, and tools that can be used to enhance privacy in the context of Coinswap.
+
+## Traceability
+
+Bitcoin transactions are inherently traceable due to the public nature of the blockchain. When coins are swapped using Coinswap, the transaction history of the coins involved remains visible on the blockchain.
+
+By using the Coinswap protocol, the transaction graph is obfuscated, making it harder to trace the flow of coins between participants. This obfuscation is achieved through the use of time-locked contracts, hash preimages, and private key handovers, which ensure that the swap process is atomic and private.
```
[the remark this answers is not in the document]
Comment (rishkwal) [reply v1/6_privacy.md]: Removed the file, will create an issue for it

**Pull request merged**